### DISCUSSION TITLE
How to detect floors or walls?

### DISCUSSION THREAD

**Discussion by sergiofigueras** [google/model-viewer#2011] [Q&A]
Hello there, community!

So, I'm building an app in which I'm going to paint a wall or a floor, after its detection.

Do you see that it could work, somehow? I was thinking in a flow like:
1 - Listen to detect floor or wall events
2 - Trigger an event that will build a USDz or GLB on the client-side, based on the platform which it's running.

Any thoughts? All are highly appreciated :)

**Comment by elalish** [comment]
That's a bit lower-level than model-viewer is really designed for. I think you'd be better off using three.js directly and their interface to WebXR.
